src/HealthGPS.Console/program.cpp

```diff
@@ -156,9 +158,15 @@ int main(int argc, char *argv[]) {
         std::this_thread::sleep_for(std::chrono::milliseconds(200));
         fmt::print(fg(fmt::color::light_green), "\nCompleted, elapsed time : {}ms\n\n", runtime);
         event_monitor.stop();
+
+#ifdef CATCH_EXCEPTIONS
     } catch (const std::exception &ex) {
         fmt::print(fg(fmt::color::red), "\n\nFailed with message: {}.\n\n", ex.what());
+
+        // Rethrow exception so it can be handled by OS's default handler
+        throw;
```

Comment:
Is it better to call `std::rethrow_exception (ex)` or just throw in this case. Not a nitpick, genuine question!